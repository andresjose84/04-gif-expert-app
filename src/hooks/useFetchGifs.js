import { useEffect, useState } from "react"
import { getGif } from '../helpers/getGif';

export const useFetchGifts = (category) => {
    const [state, setState] = useState({
        data : [],
        loading: true
    });

    useEffect( () => {
        getGif( category ).then(obj => {
            setState({
                data : obj,
                loading: false
            });
        })
    }, [category])

    return state;
}