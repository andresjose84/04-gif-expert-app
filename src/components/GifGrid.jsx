import React from 'react';
import PropTypes from 'prop-types'
import GifGridItem from './GifGridItem';
import { useFetchGifts } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifts(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>

            { loading && <p className='animate__animated animate__flash'>Loading...</p> }

            <div className='card-grid'>
                {
                    images.map(ele => (
                        <GifGridItem 
                        key={ ele.id } 
                        { ...ele }
                        />
                        ))
                    }
            </div> 
        </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}

export default GifGrid