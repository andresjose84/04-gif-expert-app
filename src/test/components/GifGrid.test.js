import {React} from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { useFetchGifts } from '../../hooks/useFetchGifs';
import GifGrid from '../../components/GifGrid';
jest.mock('../../hooks/useFetchGifs');


describe('Validamos el componente <GifGrid />', () => { 
    test('debe de generar el snapshot', () => { 
        useFetchGifts.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow( <GifGrid category="killua"/>);
        expect(wrapper).toMatchSnapshot();   
    });
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => { 
        const obj = [{
            id : 1,
            url: 'https://localhost/asdfafdssfd/img.jpg',
            title: 'cualquier cosa'
        }]
        useFetchGifts.mockReturnValue({
            data: obj,
            loading: true
        });
        const wrapper = shallow( <GifGrid category="killua"/>);

        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(true);
        expect(wrapper.find('GifGridItem').length).toBe(obj.length)
    });
})