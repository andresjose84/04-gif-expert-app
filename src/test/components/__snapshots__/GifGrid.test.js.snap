// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validamos el componente <GifGrid /> debe de generar el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    killua
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
